### ISSUE TITLE
Prevent duplicate installs by handling missing OAUTH activitation.

### ISSUE THREAD

**Issue by tonesto7** [tonesto7/nest-manager#23]

**Comment by tonesto7** [comment]
I still need to address this.  I will make sure it's in the next release

**Comment by tonesto7** [comment]
This is currently in the latest dev version

**Comment by imnotbob** [comment]
I think singleInstance: true should be added to the smartapp, this would prevent multiple installs in the location

**Comment by tonesto7** [comment]
I don't agree... that would prevent the app from being installed more than once for multiple locations.

The multiple install issue has been corrected.  And it will no longer fail with oauth being off. I will tell the user that oauth is not enabled in the app

**Comment by imnotbob** [comment]
I don’t think you are interpretting the docs correctly.  singleInstance is per location.  I have multiple instances of the app installed, one at each location with singleInstance on for both.

To be clear:

  If customer has 1 ST hub, - you are correct
  if customer has 2 ST hubs, 2 locations, then it works as I describe.

It is not clear to me that we should enable a ST hub to control things outside of the hub’s location.   I can see where there would be different opinions, but when you write automations, the world quickly crashes down.

For example - you could not use our automations, as we could not select other devices (in the other location) to use with the local running automation.

In short singleInstance: true should be our default.  If folks want to do strange things, let them be on their own.

Eric

On Apr 13, 2016, at 7:38 PM, Tonesto7 <notifications@github.com<mailto:notifications@github.com>> wrote:

I don't agree... that would prevent the app from being installed more than once for multiple locations.

The multiple install issue has been corrected. And it will no longer fail with oauth being off. I will tell the user that oauth is not enabled in the app

**Comment by tonesto7** [comment]
I fully understand how it works and that it's per ST location.  I am referring to the users who have more than one Nest Structure.  If they wanted to use both they would need 2 installs of the SmartApp.  If singleInstance is set this would not be possible.

**Comment by imnotbob** [comment]
I do understand.  But once they do what you describe, the automations we have created won’t work, as there will be no sensors at the other location, or modes at the other location.

I’m asking why do we want a single ST location to control two nest locations?  It seems like nothing but pain will result..

On Apr 13, 2016, at 7:50 PM, Tonesto7 <notifications@github.com<mailto:notifications@github.com>> wrote:

I fully understand how it works and that it's per ST location. I am referring to the users who have more than one Nest Structure. If they wanted to use both they would need 2 installs of the SmartApp. If singleInstance is set this would not be possible.

**Comment by tonesto7** [comment]
I've been thinking about this.  I can see what you are referring too but what about users who have nests and more than one home but only have an ST hub at their primary location.

**Comment by imnotbob** [comment]
So they only have home automation in one location.

They use the Nest app for the other until they decide to do HA at their second location.

On Apr 14, 2016, at 10:25 AM, Tonesto7 <notifications@github.com<mailto:notifications@github.com>> wrote:

I've been thinking about this. I can see what you are referring too but what about users who have nests and more than one home but only have an ST hub at their primary location.

**Comment by tonesto7** [comment]
You will see the singleInstance set on the next commit
